### ISSUE TITLE
Unsuccessful cross-attention weight loading in Custom Diffusion with StableDiffusionAttendAndExcitePipeline

### ISSUE THREAD

**Issue by daeunni** [huggingface/diffusers#7286]
### Describe the bug

I load the trained pytorch_custom_diffusion_weights.bin using StableDiffusionAttendAndExcitePipeline, however, loading is failed. I am wondering if is it the same cause with this issue (https://github.com/huggingface/diffusers/issues/7261#issuecomment-1990962268) .

### Reproduction

pipeline = StableDiffusionAttendAndExcitePipeline.from_pretrained(
    "CompVis/stable-diffusion-v1-4", torch_dtype=torch.float16,
).to("cuda")

cur_ckpt = '/d1/daeun/diffusers/examples/custom_diffusion/exp/cyw_step300'
pipeline.unet.load_attn_procs(cur_ckpt, #+ '/checkpoint-300', 
                                # weight_name = 'model.safetensors'
                                weight_name="pytorch_custom_diffusion_weights.bin"
                                )
pipeline.load_textual_inversion(cur_ckpt + "/<new1>.bin", weight_name="<new1>.bin")
pipeline.load_textual_inversion(cur_ckpt + "/<new2>.bin", weight_name="<new2>.bin")
pipeline.load_textual_inversion(cur_ckpt + "/<new3>.bin", weight_name="<new3>.bin")
pipeline.load_textual_inversion(cur_ckpt + "/<new4>.bin", weight_name="<new4>.bin")
pipeline.load_textual_inversion(cur_ckpt + "/<new5>.bin", weight_name="<new5>.bin")

pipeline.scheduler = DDPMScheduler.from_config(pipeline.scheduler.config)
generator = torch.Generator(device="cuda")
prompt = "a <new3> cat playing with a <new2> duck toy in the grassland"
image = pipeline(
    prompt,
    token_indices=[3, 8], #token_indices,
    num_inference_steps=100,
    guidance_scale=6.0,
    generator = generator,  
    eta=1.0
).images[0]

### System Info

- `diffusers` version: 0.26.0.dev0
- Platform: Linux-4.15.0-99-generic-x86_64-with-glibc2.17
- Python version: 3.8.18
- PyTorch version (GPU?): 2.2.0+cu118 (True)
- Huggingface_hub version: 0.20.3
- Transformers version: 4.37.2
- Accelerate version: 0.26.1
- xFormers version: 0.0.24+cu118
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

@sayakpaul

**Comment by sayakpaul** [comment]
Can we try to reduce the reproduction code? 

Also, I don't see the error trace.

**Comment by Rbrq03** [comment]
hey, maybe #7294 helps

**Comment by sayakpaul** [comment]
Closing because of inactivity.
